pkg/virtctl/adm/logverbosity/logverbosity.go

```diff
@@ -0,0 +1,443 @@
+	var componentNameToJSONName = map[string]string{
+		"virt-api":        "virtAPI",
+		"virt-controller": "virtController",
+		"virt-handler":    "virtHandler",
+		"virt-launcher":   "virtLauncher",
+		"virt-operator":   "virtOperator",
+		"all":             "all",
+	}
+	return componentNameToJSONName[componentName]
+}
+
+func detectInstallNamespaceAndName(virtClient kubecli.KubevirtClient) (namespace, name string, err error) {
+	// listing KubeVirt CRs in all namespaces and see where it is installed
+	kvs, err := virtClient.KubeVirt("").List(&k8smetav1.ListOptions{})
+	if err != nil {
+		return "", "", fmt.Errorf("could not list KubeVirt CRs in all namespaces: %v", err)
+	}
+	if len(kvs.Items) == 0 {
+		return "", "", errors.New("could not detect a KubeVirt installation")
+	}
+	if len(kvs.Items) > 1 {
+		return "", "", errors.New("invalid kubevirt installation, more than one KubeVirt resource found")
+	}
+	namespace = kvs.Items[0].Namespace
+	name = kvs.Items[0].Name
+	return
+}
+
+func hasVerbosityInKV(kv *v1.KubeVirt) (map[string]uint, error) {
+	verbosityMap := map[string]uint{} // key: component name, value: verbosity
+
+	// if LogVerbosity is not present in the KubeVirt CR, skip here
+	if kv.Spec.Configuration.DeveloperConfiguration.LogVerbosity != nil {
+		// get the logVerbosity field in the KubeVirt CR, and put it to verbosityMap
+		lvJSON, err := json.Marshal(kv.Spec.Configuration.DeveloperConfiguration.LogVerbosity)
+		if err != nil {
+			return nil, err
+		}
+		if err := json.Unmarshal(lvJSON, &verbosityMap); err != nil {
+			return nil, err
+		}
+	}
+
+	return verbosityMap, nil
+}
+
+func createOutputLines(verbosityVal map[string]uint, options map[string]uint) []string {
+	var lines []string
+	_, allIsSet := options["all"]
+	for component := VirtAPI; component < All; component++ { // all is the last component, and do not need to check it
+		componentName := GetComponentNameByVirtComponent(component)
+		JSONName := getJSONNameByVirtComponent(component)
+		if _, exist := options[componentName]; exist || allIsSet {
+			line := fmt.Sprintf("%s=%d", componentName, verbosityVal[JSONName])
+			lines = append(lines, line)
+		}
+	}
+	return lines
+}
+
+func createShowMessage(currentLv, options map[string]uint) []string {
+	// fill the unattended verbosity with default verbosity
+	// key: JSONName, value: verbosity
+	var verbosityVal = map[string]uint{
+		"virtAPI":        virtconfig.DefaultVirtAPILogVerbosity,
+		"virtController": virtconfig.DefaultVirtControllerLogVerbosity,
+		"virtHandler":    virtconfig.DefaultVirtHandlerLogVerbosity,
+		"virtLauncher":   virtconfig.DefaultVirtLauncherLogVerbosity,
+		"virtOperator":   virtconfig.DefaultVirtOperatorLogVerbosity,
+	}
+
+	// update the verbosity based on the existing verbosity in the KubeVirt CR
+	for key, value := range currentLv {
+		verbosityVal[key] = value
+	}
+
+	// create a message to show verbosity for the specified component
```

Comment:
nit: superfluous comment, it is clear from the function name

Reply:
Agreed. Will remove it.